homeassistant/components/airzone/sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for the Airzone sensors."""
+from __future__ import annotations
+
+from typing import Final
+
+from aioairzone.const import AZD_HUMIDITY, AZD_NAME, AZD_TEMP, AZD_ZONES
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AirzoneEntity
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+
+SENSOR_TYPES: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        key=AZD_TEMP,
+        name="Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        device_class=DEVICE_CLASS_HUMIDITY,
+        key=AZD_HUMIDITY,
+        name="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        zone_name = zone_data[AZD_NAME]
+
+        for description in SENSOR_TYPES:
+            if description.key in zone_data:
+                sensors.append(
+                    AirzoneSensor(
+                        coordinator,
+                        description,
+                        entry.entry_id,
+                        zone_id,
+                        zone_name,
+                    )
+                )
+
+    async_add_entities(sensors)
+
+
+class AirzoneSensor(AirzoneEntity, SensorEntity):
+    """Define an Airzone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        description: SensorEntityDescription,
+        entry_id: str,
+        zone_id: str,
+        zone_name: str,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, zone_id, zone_name)
+        self._attr_name = f"{zone_name} {description.name}"
+        self._attr_unique_id = f"{entry_id}_{zone_id}_{description.key}"
```

Comment:
Do you need the `system_id` in here as well?

Reply:
No, zone_id already contains the system ID:
https://github.com/Noltari/aioairzone/commit/8e3a12142b6320fdb0757289175443bf2738ae73
`zones[f"{system_id}:{zone_id}"] = zone.data()`

I introduced that change when @tkdrob requested me to remove the triple loop in aba862e90d13268e4ce68a3f919a15c7908f8e67